internal/client/client.go

```diff
@@ -453,6 +453,31 @@ func (c *CSAPI) MustDoFunc(t *testing.T, method string, paths []string, opts ...
 	return res
 }
 
+// MustDoFuncUntil is the same as MustDoFunc but retries up to 5 times, with 500 ms intervals, if the condition doesn't match.
+func (c *CSAPI) MustDoFuncUntil(t *testing.T, method string, paths []string, checkFunc func(*http.Response) bool, opts ...RequestOpt) *http.Response {
+	t.Helper()
+	var res *http.Response
+	res = nil
+	for retryNum := 0; retryNum < 5; retryNum++ {
+		res = c.DoFunc(t, method, paths, opts...)
+		if res.StatusCode < 200 || res.StatusCode >= 300 {
+			defer res.Body.Close()
+			body, _ := ioutil.ReadAll(res.Body)
+			t.Fatalf("CSAPI.MustDoFuncUntil response return non-2xx code: %s - body: %s", res.Status, string(body))
+			return res
+		}
```

Comment:
Two points of confusion:

1. I think this means you can't use `MustDoFuncUntil` to repeat a successful call until it fails. E.g. imagine a test where:
  - join a room
  - someone bans you
  - repeatedly try to change your name in the room until the ban takes place, at which point the name change fails.
2. I think this means that you can't use `MustDoFuncUntil` to repeat a failing call until it suceeds. E.g. imagine a test where:
   - fail to join a restricted room because you're not in the right space
   - join the space
   - repeatedly join the restricted room until it succeeds.

Though thinking about it some more, perhaps those ought be different functions e.g. some sort of `DoFuncUntil`.

Reply:
That's fair; not sure what better name to put on it (I've tried to keep the names consistent with what's there; I'd personally say those names aren't very intuitive either but they have docstrings).
At the very least, describing your caveats in the docstring would probably help though.